Regex.po

nabilbendafi opened the pull request:
Partial translation for #118

JulienPalard commented on howto/regex.po:
```diff
@@ -7,19 +7,19 @@ msgid ""
 #: ../Doc/howto/regex.rst:5
 msgid "Regular Expression HOWTO"
-msgstr ""
+msgstr "HOWTO Expressions régulières"
```
I would translate "howto", maybe "Guide des expression régulières" ?

nabilbendafi replied:
Done

JulienPalard commented on howto/regex.po:
```diff
@@ -32,160 +32,250 @@ msgstr ""
 "This document is an introductory tutorial to using regular expressions in "
-"Python with the :mod:`re` module.  It provides a gentler introduction than "
-"the corresponding section in the Library Reference."
+"Python with the :mod:`re` module.  It provides a gentler introduction than the "
```
Please configure your po editor to avoid changing msgids. Using "poindent" from pypi may help sometimes reformatting your po file before submitting them. having diff in msgids make git diffs hard to read.

JulienPalard commented on howto/regex.po:
```diff
@@ -32,160 +32,250 @@ msgstr ""
+"Python with the :mod:`re` module.  It provides a gentler introduction than the "
+"corresponding section in the Library Reference."
 msgstr ""
+"Ce document est un tutoriel d'introduction à l'utilisation des expressions "
```
Maybe use "guide" instead of "tutoriel" if we choose "guide" in place of "howto" for consistency.

nabilbendafi replied:
Done

JulienPalard commented on howto/regex.po:
```diff
@@ -32,160 +32,250 @@ msgstr ""
+"\".  You can also use REs to modify a string or to split it apart in various "
+"ways."
+msgstr ""
+"Les expressions régulières (appelées REs, ou regexes, ou motifs regex) sont "
```
Please also quote "expressions rationnelles", widely used by non-users of "expression régulières". I would also split "motifs" and "regex", I *never* heard of "motifs regex", yet I heard of "regex" and wikipedia heard of "motifs".

JulienPalard commented on howto/regex.po:
```diff
@@ -32,160 +32,250 @@ msgstr ""
+msgstr ""
+"Les expressions régulières (appelées REs, ou regexes, ou motifs regex) sont "
+"essentiellement un petit langage de programmation hautement spécialisé embarqué "
+"dans Python et rendu disponible au travers du module :mod:`re. En utilisant ce "
```
Missing backtick after `re`.

nabilbendafi replied:
Done

JulienPalard commented on howto/regex.po:
```diff
@@ -32,160 +32,250 @@ msgstr ""
+"Nous commencerons par étudier les expressions régulières les plus simples. Dans "
+"le mesure où les expressions régulières sont utilisées pour manipuler des "
+"chaînes de caractères, nous commencerons par les tâches les plus communes: "
+"caractères ressemblants."
```
"ressemblants" is badly choosen here, semantically we're just trying to express, by matching characters, that "toto" matches "toto".

nabilbendafi replied:
Done

JulienPalard commented on howto/regex.po:
```diff
@@ -32,160 +32,250 @@ msgstr ""
+"enable a case-insensitive mode that would let this RE match ``Test`` or "
 "``TEST`` as well; more about this later.)"
 msgstr ""
+"La plupart des lettres ou caractères vont correspondre à eux-mêmes . Par "
```
Please remove the space before the dot.

nabilbendafi replied:
Done

JulienPalard commented on howto/regex.po:
```diff
@@ -32,160 +32,250 @@ msgstr ""
+"La plupart des lettres ou caractères vont correspondre à eux-mêmes . Par "
+"exemple, l'expression régulière ``test`` correspond à la chaîne de caractère "
+"``test`` précisément. (Vous pouvez activer le mode non-sensible à la casse qui "
+"permet à cette RE de correspondre à ``Test`` ou ``TEST`` également; plus à ce "
```
"plus à ce sujet dans la suite" looks just strange.

JulienPalard commented on howto/regex.po:
```diff
@@ -32,160 +32,250 @@ msgstr ""
+"the RE by repeating them or changing their meaning.  Much of this document is "
+"devoted to discussing various metacharacters and what they do."
+msgstr ""
+"Il y a des exceptions à cette règle; certains caractères sont spéciaux :dfn:"
```
"sont spéciaux metacharacters" does not look french at all.

nabilbendafi replied:
Done

christopheNan commented on howto/regex.po:
```diff
@@ -32,160 +32,250 @@ msgstr ""
+"`metacharacters`, et ne correspondent pas à eux-mêmes. Au lieu de cela, ils "
+"signalent que certaines choses non ordinaires doivent correspondre, ou "
+"affectent d'autre portions de la RE en les répétant ou en changeant leur sens. "
+"La majorité de ce document oriente la discussion autour des différents méta-"
```
Une grande partie de ce document est consacrée au fonctionnement de ces métacaractères.

nabilbendafi replied:
Done

christopheNan commented on howto/regex.po:
```diff
@@ -32,160 +32,250 @@ msgstr ""
 "characters.  If you wanted to match only lowercase letters, your RE would be "
 "``[a-z]``."
 msgstr ""
+"Les premiers méta-caractères que nous observerons sont ``[`` et ``]``. Ils sont "
```
s/observerons/étudions/

nabilbendafi replied:
Done

christopheNan commented on howto/regex.po:
```diff
@@ -32,160 +32,250 @@ msgstr ""
 "need to match a ``[`` or  ``\\``, you can precede them with a backslash to "
 "remove their special meaning: ``\\[`` or ``\\\\``."
 msgstr ""
+"Le méta-caractère qui est probablement le plus important est le **backslash**, "
```
la barre oblique inversée ?

nabilbendafi replied:
Allowed by glossary.

christopheNan commented on howto/regex.po:
```diff
@@ -32,160 +32,250 @@ msgstr ""
+"of characters that are often useful, such as the set of digits, the set of "
+"letters, or the set of anything that isn't whitespace."
 msgstr ""
+"Certaines séquences spéciales commençant par ``'\\'``représente des ensembles "
```
s/représente/représentent/
s/tel que/tels que/

nabilbendafi replied:
Done

christopheNan commented on howto/regex.po:
```diff
@@ -32,160 +32,250 @@ msgstr ""
+"pouvez utiliser la définition plus restrictive de ``\\w`` dans un motif de "
+"chaîne de caractère en spécifiant le fanion :const:`re.ASCII` lors de la "
+"compilation de l'expression régulière."
 
```
n'importe quel caractère (singulier)
Si le motif est exprimé dans le type "bytes" .... s'il est exprimé dans le type "str"
s/Unicide/Unicode/
chaîne de caractère**s**

christopheNan commented on howto/regex.po:
```diff
@@ -32,160 +32,250 @@ msgstr ""
+"standard. En générale, les versions d'Unicode trouve une correspondance avec "
+"n'importe quel caractère présent dans le catégorie appropriée de la base de "
+"données Unicode. "
 
```
s/En générale/En général/

nabilbendafi replied:
Done

christopheNan commented on howto/regex.po:
```diff
@@ -194,6 +284,8 @@ msgstr "``\\d``"
 msgstr ""
+"Correspond à n'importe quel caractère numérique; ceci est équivalent à la "
+"classe ``[0-9]``."
 
```
Les ";" doivent être précédés d'une espace insécable.

nabilbendafi replied:
Done

christopheNan commented on howto/regex.po:
```diff
@@ -233,82 +331,98 @@ msgid ""
+"will match even a newline.  ``.`` is often used where you want to match \"any "
+"character\"."
 msgstr ""
+"Le méta-caractère dans cette section est ``.``. Il correspond à tout les "
```
Le **dernier** méta-caractère

nabilbendafi replied:
Done

christopheNan commented on howto/regex.po:
```diff
@@ -233,82 +331,98 @@ msgid ""
+"``a[bcd]*b``. Elle correspond à la lettre ``'a'``, suivi d'aucune ou plusieurs "
+"lettres de la classe `[bcd]``, et finira par un ``'b'``. Maintenant supposé que "
+"cette RE correspondra à la chaîne de caractères ``'abcbd'``."
 
```
Je propose :
Détaillons un exemple étape par étape pour éclaircir tout ceci. Considérons l'expression .... Elle correspond à la lettre 'a', suivie d'aucune ou plusieurs lettres de la classe [bcd] et qui se termine par un 'b'. Maintenant, supposons que nous testions cette RE avec la chaîne 'abcbd'

christopheNan commented on howto/regex.po:
```diff
@@ -382,92 +498,92 @@ msgstr "6"
 #: ../Doc/howto/regex.rst:255
 msgid "Using Regular Expressions"
-msgstr ""
+msgstr "Utilisez les Expressions Régulières"
```
Je mettrais plutôt des substantifs dans les titres : "Utilisation des ...", "Compilation ..."

nabilbendafi replied:
Done

christopheNan commented on howto/regex.po:
```diff
@@ -584,44 +698,48 @@ msgstr ""
+"qu'en faites-on ? Les objets motifs ont plusieurs méthodes et attributs. Seuls "
+"les plus significatifs seront couverts ici; consulté la documentation :mod:"
+"`re`pour la liste complète."
 
```
Une fois que nous avons ...., qu'en faisons-nous ? ... Seuls les plus significatifs sont couverts ici ; consultez ...

nabilbendafi replied:
Done

christopheNan commented on howto/regex.po:
```diff
@@ -897,63 +1016,66 @@ msgstr ""
+"(``^`` and ``$`` haven't been explained yet;  they'll be introduced in section :"
+"ref:`more-metacharacters`.)"
 msgstr ""
+"(``^`` et ``$`` n'ont pas encore été expliqué; ils seront introduit dans le "
```
expliqués (au pluriel) ; ils sont introduits

nabilbendafi replied:
Done

nabilbendafi commented on howto/regex.po:
```diff
@@ -7,19 +7,19 @@ msgid ""
 #: ../Doc/howto/regex.rst:5
 msgid "Regular Expression HOWTO"
-msgstr ""
+msgstr "HOWTO Expressions régulières"
```
Je vais rester sur `guide`

nabilbendafi commented on howto/regex.po:
```diff
@@ -32,160 +32,250 @@ msgstr ""
 msgstr ""
+"Ce document est un tutoriel d'introduction à l'utilisation des expressions "
+"régulières en Python avec le module :mod:`re`. Il fournit une introduction plus "
+"abordable que la section correspondante dans la Bibliothèque de Référence"
```
Done

nabilbendafi commented on howto/regex.po:
```diff
@@ -32,160 +32,250 @@ msgstr ""
+msgstr ""
+"Les expressions régulières (appelées REs, ou regexes, ou motifs regex) sont "
+"essentiellement un petit langage de programmation hautement spécialisé embarqué "
+"dans Python et rendu disponible au travers du module :mod:`re. En utilisant ce "
```
L'expression `et dont la manipulation est rendue possible par l'utilisation du module` parait-elle plus adaptée ?

nabilbendafi commented on howto/regex.po:
```diff
@@ -32,160 +32,250 @@ msgstr ""
+"you have a good understanding of the matching engine's internals."
+msgstr ""
+"Les motifs d'expressions régulières sont compilés en une série de **bytecodes** "
+"qui sont ensuite exécuté par un moteur de correspondance écrit en C. Pour une "
```
Done

nabilbendafi commented on howto/regex.po:
```diff
@@ -32,160 +32,250 @@ msgstr ""
 "refer to almost any textbook on writing compilers."
 msgstr ""
+"Pour une explication détaillée sur la technologie sous-jacente des expressions "
+"régulières (automate d'état déterministe et non-déterministe), faites référence "
```
Done

JulienPalard commented:
Tes commits ont un `[WIP]` mais ta PR n'en as pas (en passant, le contraire est plus pratique, car tu peux renommer ta PR pour y enlever le `[WIP]`). Peut-on merger ou souhaite-tu continuer sur ce fichier ?

(Sachant que tu peux ausis continuer sur le fichier après le merge, sur une autre PR.)

nabilbendafi commented:
@JulienPalard I squashed my commits, and renamed it. I'd rather let you merge and close this PR. I'll open another PR whenever continuing on translating this file.
Thx
